konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/data/helpers/ProtobufUtils.java

```diff
@@ -126,14 +261,74 @@ public static Data convertProtobufToData(DataProtoMessage.DataMap dataMap) {
+                    retData.putListInt64(entry.getKey(), item.getListValue().getIList().getListList());
+                }
+                if (item.getListTypeValue() == DataProtoMessage.DataScheme.ValueType.STRING.ordinal()) {
+                    retData.putListString(entry.getKey(), item.getListValue().getSList().getListList());
+                }
+                if (item.getListTypeValue() == DataProtoMessage.DataScheme.ValueType.IMAGE.ordinal()) {
+                    List<DataProtoMessage.Image> pbImages = item.getListValue().getImList().getListList();
+                    List<Image> images = new ArrayList<>();
+                    for (val pbImage : pbImages) {
+                        List<ByteString> pbData = pbImage.getDataList();
+                        byte[] data = pbData.get(0).toByteArray();
+                        // TODO: obviously should be working for different formats
+                        Png png = new Png(data);
+                        images.add(Image.create(png));
+                    }
+                    retData.putListImage(entry.getKey(), images);
+                }
+                if (item.getListTypeValue() == DataProtoMessage.DataScheme.ValueType.NDARRAY.ordinal()) {
+                    List<DataProtoMessage.NDArray> pbArrays = item.getListValue().getNdList().getListList();
+                    List<NDArray> arrays = new ArrayList<>();
+                    for (val pbArray : pbArrays) {
+                        List<Long> shapes = pbArray.getShapeList();
+                        long[] aShapes = new long[shapes.size()];
+                        for (int i = 0; i < shapes.size(); ++i) {
+                            aShapes[i] = shapes.get(i);
+                        }
+
+                        List<ByteString> data = pbArray.getArrayList();
+                        DataProtoMessage.NDArray.ValueType type = pbArray.getType();
+                        byte[] bytes = data.get(0).toByteArray();
+                        ByteBuffer bb = ByteBuffer.wrap(bytes);
+                        // TODO: fix ndarray type
+                        SerializedNDArray ndArray = new SerializedNDArray(NDArrayType.FLOAT, aShapes, bb);
+                        arrays.add(NDArray.create(ndArray));
+                    }
+                    retData.putListNDArray(entry.getKey(), arrays);
+                }
             }
             if (item.getTypeValue() == DataProtoMessage.DataScheme.ValueType.IMAGE.ordinal()) {
                 DataProtoMessage.Image pbImage = item.getImValue();
                 List<ByteString> pbData = pbImage.getDataList();
                 byte[] data = pbData.get(0).toByteArray();
-                retData.put(entry.getKey(), Image.create(data));
+                // TODO: obviously should be working for different formats
+                Png png = new Png(data);
+                retData.put(entry.getKey(), Image.create(png));
```

Comment:
Let's add a `string type` field to protobuf, and then check it's "PNG" here (exception if it's anything else for now)
```
message Image {
    string type = 1;
    repeated bytes data = 2;
}
```

Reply:
Fixed.